pkgs/tools/typesetting/typst/default.nix

```diff
@@ -0,0 +1,53 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, stdenv
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "typst";
+  version = "unstable-2023-03-21";
+
+  src = fetchFromGitHub {
+    owner = "typst";
+    repo = "typst";
+    rev = "9fc432a41d9970dfff360fec225a926b5ab8f2ab";
+    hash = "sha256-fHKBPmq0YNNElnT6eM8zFFhrvRFReqcuqF8B5tCzaso=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "biblatex-0.6.3" = "sha256-TfH2tk7B61HHqpdGY48TdPKelp4+78x+8LRFobAg8QA=";
+      "comemo-macros-0.1.0" = "sha256-zg056kUc8sVLQ8vvT4uOuRJnyrCORsGYUvsjBJEkFPg=";
+      "hayagriva-0.1.1" = "sha256-HGQ+jNAnejxUBQNaqXPw57zfAC3qNXSWUVzDALZTXg0=";
+      "iai-0.1.1" = "sha256-EdNzCPht5chg7uF9O8CtPWR/bzSYyfYIXNdLltqdlR0=";
+      "lipsum-0.8.2" = "sha256-deIbpn4YM7/NeuJ5Co48ivJmxwrcsbLl6c3cP3JZxAQ=";
+      "pixglyph-0.1.0" = "sha256-8veNF3rzV21ayzk9gh2x0mQA8nHGM662ohvh084a0vk=";
+      "unicode-math-class-0.1.0" = "sha256-NkwDzj1SfUe570UcfotmVP6bIEYwiegZd0j8TPEWoOk=";
+    };
+  };
```

Comment:
```suggestion
  cargoSha256 = "sha256-aEAj0Lhr/1/kPCgmiv+Nc6y0LTUuLJSbNO13Ou5qBDI=";
```

Reply:
Don't forget to remove lock file itself too!

Reply:
this is intentional: https://github.com/NixOS/nixpkgs/pull/221716